### PR TITLE
[Compute] SpotPlacementRecommender resource split

### PR DESCRIPTION
**Pull request by NoriZC**
## Description
Based  on https://github.com/Azure/azure-powershell/pull/25361 @Elina-L 
Add alias for `Invoke-AzSpotPlacementRecommender` to avoid breaking changes.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by Elina-L** [comment]
ChangeLog:

Adding "Invoke-AzSpotPlacementScore" cmdlet. This cmdlet calls the latest Spot Placement Score API [https://management.azure.com/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{region}/**placementScores/spot/generate](https://management.azure.com/subscriptions/%7BsubscriptionId%7D/providers/Microsoft.Compute/locations/%7Bregion%7D/**placementScores/spot/generate)**
Remove the "Invoke-AzSpotPlacementRecommender" cmdlet. This cmdlet calls the soon to be deprecated Spot Placement Recommender API: [https://management.azure.com/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{region}/**diagnostics/spotPlacementRecommender/generate](https://management.azure.com/subscriptions/%7BsubscriptionId%7D/providers/Microsoft.Compute/locations/%7Bregion%7D/**diagnostics/spotPlacementRecommender/generate)*

File: src/Compute/Compute.Autorest/UX/Microsoft.Compute/virtualMachineScaleSets-virtualMachines-runCommands.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "resourceType": "virtualMachineScaleSets/virtualMachines/runCommands",
-  "apiVersion": "2021-07-01",
+  "apiVersion": "2023-07-01",
```
Comment (Sandido) [inline src/Compute/Compute.Autorest/UX/Microsoft.Compute/virtualMachineScaleSets-virtualMachines-runCommands.json:3]: Why not the latest version? 2024-03-01.

**Review by Sandido — APPROVED** [review]

File: src/Compute/Compute.Autorest/docs/Get-AzGalleryApplicationVersion.md [hunk 2]
```diff
@@ -53,7 +53,8 @@ Retrieve all the Gallery Application Version resources in the provided Resource
 ## PARAMETERS
 
 ### -DefaultProfile
-The credentials, account, tenant, and subscription used for communication with Azure.
```
Comment (Sandido) [inline src/Compute/Compute.Autorest/docs/Get-AzGalleryApplicationVersion.md:56]: How do you know this?

**Review by Sandido — APPROVED** [review]

File: src/Compute/Compute.Autorest/UX/Microsoft.Compute/virtualMachineScaleSets-virtualMachines-runCommands.json (hunk 1 above)
Comment (grizzlytheodore) [reply src/Compute/Compute.Autorest/UX/Microsoft.Compute/virtualMachineScaleSets-virtualMachines-runCommands.json:3]: the generated cmdlets have not migrated to the newest version yet as there has not been a need to. 

The readme files that are used to generated cmdlets still point to 2023-07-01, and these files are agnostic to the newer version

File: src/Compute/Compute.Autorest/docs/Remove-AzGalleryApplication.md [hunk 3]
```diff
@@ -55,7 +55,8 @@ Accept wildcard characters: False
 ```
 
 ### -DefaultProfile
-The credentials, account, tenant, and subscription used for communication with Azure.
+The DefaultProfile parameter is not functional.
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Autorest/docs/Remove-AzGalleryApplication.md:58]: Adam made a comment on these in chat, but where are these gallery related cmdlet changes coming from? 
Can we leave them as they are ? Was it suggested by the PowerShell team?

**Comment by Elina-L** [comment]
Continuing discussion on this PR: https://github.com/Azure/azure-powershell/pull/25361
